app/Library/UserService.php

```diff
@@ -17,7 +17,7 @@ public function findOrCreateByEmplid(string $emplid): ?User {
             return $user;
         }
 
-        $user = LDAP::lookupUser($emplid, 'umnemplid');
+        $user = LDAP::lookupUser(str_pad($emplid, 7, 0, STR_PAD_LEFT), 'umnemplid');
```

Comment:
Could you explain a little why this is needed? Does it handle some special case?

Reply:
some emplids start with leading zeros, and LDAP treats them as strings.

Reply:
Makes sense, thanks!